File: agents/purfle-pet/csharp/AsciiArt.cs
namespace PurflePet;

/// <summary>
/// ASCII art faces for each pet mood.
/// </summary>
public static class AsciiArt
{
    public static string Happy()    => @"(^‿^)";
    public static string Sad()      => @"(╥_╥)";
    public static string Hungry()   => @"(>_<)~";
    public static string Sleepy()   => @"(o_o)zzZ";
    public static string Excited()  => @"\(★ω★)/";

    /// <summary>
    /// Returns the ASCII art face for the given mood string.
    /// </summary>
    public static string ForMood(string mood) => mood.ToLowerInvariant() switch
    {
        "happy"   => Happy(),
        "sad"     => Sad(),
        "hungry"  => Hungry(),
        "sleepy"  => Sleepy(),
        "excited" => Excited(),
        _         => Happy()
    };
}

File: agents/purfle-pet/typescript/src/ascii-art.ts
/**
 * ASCII art faces for each pet mood.
 */

export function happy(): string {
  return "(^‿^)";
}

export function sad(): string {
  return "(╥_╥)";
}

export function hungry(): string {
  return "(>_<)~";
}

export function sleepy(): string {
  return "(o_o)zzZ";
}

export function excited(): string {
  return "\\(★ω★)/";
}

/**
 * Returns the ASCII art face for the given mood string.
 */
export function forMood(mood: string): string {
  switch (mood.toLowerCase()) {
    case "happy":
      return happy();
    case "sad":
      return sad();
    case "hungry":
      return hungry();
    case "sleepy":
      return sleepy();
    case "excited":
      return excited();
    default:
      return happy();
  }
}
